nemo/collections/nlp/models/language_modeling/megatron_bert_model.py

```diff
@@ -91,8 +104,48 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
             add_binary_head=cfg.bert_binary_head,
             megatron_legacy=cfg.get('megatron_legacy', False),
         )
-        # not using amp o2
-        self.megatron_amp_o2 = False
+
+        if self.megatron_amp_o2:
+            if isinstance(self.model, list):
```

Comment:
Can we remove the conditions that check for whether a model is a list? This should only happen for interleaved pipeline parallelism which is only supported for GPT. You can take a look at how this bit is implemented for T5.